### PR TITLE
FORGE-2226 : Being able to add a CDI injection point

### PR DESCRIPTION
**Pull request by agoncal**

### CODE REVIEW

File: javaee/tests/src/test/java/org/jboss/forge/addon/javaee/cdi/CDIOperationsTest.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+/**
+ * Unit tests to verify the behavior of the {@link CDIOperations} implementation class.
+ */
+@RunWith(Arquillian.class)
+public class CDIOperationsTest
+{
+   @Deployment
+   @AddonDependencies({
+            @AddonDependency(name = "org.jboss.forge.addon:ui-test-harness"),
```
Comment (gastaldi) [inline javaee/tests/src/test/java/org/jboss/forge/addon/javaee/cdi/CDIOperationsTest.java]: You can use just @AddonDependencies , Furnace will detect the dependencies from the tests' pom.xml

File: javaee/impl/src/main/java/org/jboss/forge/addon/javaee/cdi/CDIOperations.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+               {
+                  JavaSource<?> javaSource = resource.getJavaType();
+                  if (!javaSource.isAnnotation() && !javaSource.isEnum()
+                           // CDI
+                           && !javaSource.hasAnnotation(Decorator.class)
+                           && !javaSource.hasAnnotation(Interceptor.class)
+                           // EJB
+                           && !javaSource.hasAnnotation(MessageDriven.class)
+                           // JPA
+                           && !javaSource.hasAnnotation(Entity.class)
```
Comment (gastaldi) [inline javaee/impl/src/main/java/org/jboss/forge/addon/javaee/cdi/CDIOperations.java]: Can't we reuse these checks from the other xxxOperations classes?

File: javaee/tests/src/test/java/org/jboss/forge/addon/javaee/cdi/CDIOperationsTest.java (hunk 1 above)
Comment (agoncal) [inline javaee/tests/src/test/java/org/jboss/forge/addon/javaee/cdi/CDIOperationsTest.java]: Oh, didn't know that. Really handy

File: javaee/impl/src/main/java/org/jboss/forge/addon/javaee/cdi/CDIOperations.java (hunk 2 above)
Comment (agoncal) [inline javaee/impl/src/main/java/org/jboss/forge/addon/javaee/cdi/CDIOperations.java]: Hum.... not really. This is very specific to CDI. As an example, `EJBOperarations` has the `isEJB` method (stateless, stateful, MDB). But with CDI you can inject session beans but not MDB. So the `isEJB` method is useless in this case

**Pull request merged**